src/modules/rest/impl/src/main/java/it/geosolutions/geostore/services/rest/security/HeadersAuthenticationFilter.java

```diff
@@ -77,6 +81,9 @@ protected void authenticate(HttpServletRequest req) {
                     if (groupName.equals(GroupReservedNames.EVERYONE.groupName())) {
                         everyoneFound = true;
                     }
+                    if(groupHeaderPrefix != null && groupName.startsWith(groupHeaderPrefix)) {
```

Comment:
It would be useful to implement a more generic (and optional) groupName mapper that transforms group names using an SpEl expression.

Reply:
I thought about it, I didn't do that because testing was hard.
I need a map of regex --> expression.